src/baldr/incident_singleton.h

```diff
@@ -46,23 +48,32 @@ struct incident_singleton_t {
   // daemon thread to watch for new incident data
   std::thread watcher;
 
+  // prototype for the watch function. we need this so unit tests can safely test all functionality
+  using watch_function_t = std::function<void(boost::property_tree::ptree,
+                                              std::unordered_set<valhalla::baldr::GraphId>,
+                                              std::shared_ptr<state_t>,
+                                              std::function<bool(size_t)>)>;
+
   /**
    * Singleton private constructor that static function uses to instantiate the singleton
-   * @param config   lets the daemon thread know where/how to look for incidents
-   * @param tileset  an mmapped graph tileset (ie static) allows incident loading to be lock-free
+   * @param config      lets the daemon thread know where/how to look for incidents
+   * @param tileset     an mmapped graph tileset (ie static) allows incident loading to be lock-free
+   * @param watch_func  the function the background thread will run to keep incident caches up to date
    */
   incident_singleton_t(const boost::property_tree::ptree& config,
-                       const std::unordered_set<valhalla::baldr::GraphId>& tileset)
-      : state{new state_t{}}, watcher(watch, config, tileset, state, interrupt()) {
+                       const std::unordered_set<valhalla::baldr::GraphId>& tileset,
+                       const watch_function_t& watch_func = incident_singleton_t::watch)
```

Comment:
i also added the ability to specify a watch function so that i could reenable the constructor test and test both outcomes (one where initialization passes and one where it fails)